test/jdk/javax/swing/plaf/basic/BasicTreeUI/TreeCellRendererLeakTest.java

```diff
@@ -0,0 +1,208 @@
+            } else {
+                label.setBackground(getBackgroundNonSelectionColor());
+            }
+
+            weakRefArrLabel.add(smCount++, new WeakReference<JLabel>(label));
+            return label;
+        }
+    }
+
+    public TreeCellRendererLeakTest() {
+        initComponents();
+        jTree1.setCellRenderer(new TreeCellRenderer());
+        Thread updateThread = new Thread(this::runChanges);
+        updateThread.setDaemon(true);
+        updateThread.start();
+        Thread infoThread = new Thread(this::runInfo);
+        infoThread.setDaemon(true);
+        infoThread.start();
+    }
+
+    // <editor-fold defaultstate="collapsed" desc=" Generated Code ">
+    private void initComponents() {
+        jTabbedPane1 = new javax.swing.JTabbedPane();
+        jPanel1 = new javax.swing.JPanel();
+        jScrollPane1 = new javax.swing.JScrollPane();
+        jTree1 = new javax.swing.JTree();
+        jPanel2 = new javax.swing.JPanel();
+
+        jPanel1.setLayout(new java.awt.BorderLayout());
+
+        jScrollPane1.setViewportView(jTree1);
+
+        jPanel1.add(jScrollPane1, java.awt.BorderLayout.CENTER);
+
+        jTabbedPane1.addTab("tab1", jPanel1);
+
+        jPanel2.setLayout(new java.awt.BorderLayout());
+
+        jTabbedPane1.addTab("tab2", jPanel2);
+
+        jTabbedPane1.setSelectedIndex(1);
+
+        frame = new JFrame();
+        frame.getContentPane().add(jTabbedPane1, java.awt.BorderLayout.CENTER);
+
+        frame.pack();
+        frame.setLocationRelativeTo(null);
+        frame.setVisible(true);
+    }// </editor-fold>
+
+    public static void main(String args[]) throws Exception {
+        try {
+            testDone = new CountDownLatch(1);
+            SwingUtilities.invokeAndWait(() -> {
+                TreeCellRendererLeakTest tf = new TreeCellRendererLeakTest();
+            });
+            testDone.await();
+        } finally {
+            SwingUtilities.invokeAndWait(() -> {
+                if (frame != null) {
+                    frame.dispose();
+                }
+            });
+        }
+    }
+
+    // Periodically cause a nodeChanged() for one of the nodes
+    public void runChanges() {
+        long count = 0;
+        long time = System.currentTimeMillis();
+        long tm = System.currentTimeMillis();
+        while ((tm - time) < (15 * 1000)) {
+            final long currentCount = count;
+            try {
+                SwingUtilities.invokeAndWait(new Runnable( ) {
+                    public void run( ) {
+                        DefaultTreeModel model = (DefaultTreeModel) jTree1.getModel();
+                        TreeNode root = (TreeNode) model.getRoot();
+                        DefaultMutableTreeNode n = (DefaultMutableTreeNode) model.getChild(root, 0);
+                        n.setUserObject("runcount " + currentCount);
+                        model.nodeChanged(n);
+                    }
+                });
```

Comment:
The anonymous class can be a lambda expression.

The code inside the `Runnable` can be reduced to

```java
n.setUserObject("runcount " + currentCount);
model.nodeChanged(n);
```

This implies, you make `n` a field and initialise it in `initComponents`; or after calling `initComponents`.